ocaml/forkexecd/test/fe_test.ml

```diff
@@ -68,8 +68,24 @@ let shuffle x =
   done ;
   Array.to_list arr
 
+let fd_list () =
+  let pid = Unix.getpid () in
+  let path = Printf.sprintf "/proc/%d/fd" pid in
+  List.filter (* get rid of the fd used to read the directory *)
+    (fun x ->
+      try
+        ignore (Unix.readlink (Filename.concat path x)) ;
+        true
+      with _ -> false
+    )
+    (Array.to_list (Sys.readdir path))
```

Comment:
This creates a list only to later take its length. Can we return the array, whose length could be easily taken as well.

Reply:
It's used in another function, code was taken there and it's reused for new test also

Reply:
Okay, I added a different change instead. As one usage use the list but compute again the links (calling `Unix.readlink`) and the other just use `List.length` to get the number I changed the code to use `Array.fold_right` to avoid array -> list conversion and call `Unix.readlink` just once.